### ISSUE TITLE
False positives for SpacingBetweenDeclarationsWithAnnotationsRule

### ISSUE THREAD

**Issue by chao2zhang** [pinterest/ktlint#1125]
## Expected Behavior
```
fun foo() {
    val a

    // hello
    @Foo
    val b
}
```
should get empty result when checked by `SpacingBetweenDeclarationsWithAnnotationsRule`

## Observed Behavior
The above code snipped has an error reported by `SpacingBetweenDeclarationsWithAnnotationsRule`

## Steps to Reproduce
Use the code sample above.

## Your Environment
* Version of ktlint used: 0.41.0 / Latest main branch
